source/PlayerInfo.cpp

```diff
@@ -1634,6 +1645,37 @@ void PlayerInfo::HandleEvent(const ShipEvent &event, UI *ui)
 
 
 
+// Callback for handling the responses of a generic conversation (i.e. not one
+// that offers a mission). These might be from "on enter", or from completing
+// the conditions associated with a mission NPC. This callback essentially
+// manages the death of the player.
+void PlayerInfo::BasicCallback(int response)
+{
+	// Since no mission is being offered, the difference between accept,
+	// decline, defer and their "forced leave" counterparts launch, flee,
+	// and depart, is whether any NPC associated with the conversation is
+	// killed (by this function's caller) when the conversation ended.
+	
+	if(response == Conversation::ACCEPT || response == Conversation::LAUNCH)
+	{
+		// Nothing to do.
+	}
+	else if(response == Conversation::DECLINE || response == Conversation::FLEE)
+	{
+		// Nothing to do.
+	}
+	else if(response == Conversation::DEFER || response == Conversation::DEPART)
+	{
+		// Nothing to do.
+	}
```

Comment:
I don't think all the above conditions need to be here. To my mind it would be clearer to only have an `if` for the responses that actually have a result.

Reply:
I agree, they don't need to be here. I added them because I thought it was possibly beneficial to explicitly convey that "yes, these responses are possible, here is where they are handled, and they do nothing w.r.t the player."